File: onyx-database/src/com/onyx/diskmap/base/concurrent/EmptyMap.java
package com.onyx.diskmap.base.concurrent;

import com.onyx.util.map.AbstractCompatMap;

import java.util.Collection;
import java.util.Map;
import java.util.Set;
import java.util.function.BiConsumer;
import java.util.function.BiFunction;

/**
 * Created by tosborn1 on 2/19/17.
 * <p>
 * This class is a map that does not perform any actions.  The purpose is so that it can be injected to ignore caching
 */
public class EmptyMap extends AbstractCompatMap {
    @Override
    public int size() {
        return 0;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }

    @Override
    public boolean containsKey(Object key) {
        return false;
    }

    @Override
    public boolean containsValue(Object value) {
        return false;
    }

    @Override
    public Object get(Object key) {
        return null;
    }

    @Override
    public Object put(Object key, Object value) {
        return null;
    }

    @Override
    public Object remove(Object key) {
        return null;
    }

    @Override
    public void putAll(Map m) {}

    @Override
    public void clear() {}

    @Override
    public Set keySet() {
        return null;
    }

    @Override
    public Collection values() {
        return null;
    }

    @Override
    public Set<Entry> entrySet() {
        return null;
    }

    @Override
    public void forEach(BiConsumer action) {

    }

    @Override
    public void replaceAll(BiFunction function) {

    }

}

File: onyx-database/src/main/kotlin/com/onyx/concurrent/impl/EmptyMap.kt
package com.onyx.concurrent.impl

/**
 * Created by tosborn1 on 2/19/17.
 *
 * This class is a map that does not perform any actions.  The purpose is so that it can be injected to ignore caching
 */
class EmptyMap<K,V> : MutableMap<K, V> {
    override val size: Int
        get() = 0
    override val entries: MutableSet<MutableMap.MutableEntry<K, V>>
        get() = HashSet()
    override val keys: MutableSet<K>
        get() = HashSet()
    override val values: MutableCollection<V>
        get() = HashSet()

    override fun containsKey(key: K): Boolean = false
    override fun containsValue(value: V): Boolean = false
    override fun get(key: K): V? = null
    override fun isEmpty(): Boolean = true
    override fun clear() {}
    override fun put(key: K, value: V): V? = null
    override fun putAll(from: Map<out K, V>) {}
    override fun remove(key: K): V? = null
}
